src/Stache/Query/UserQueryBuilder.php

```diff
@@ -32,6 +32,16 @@ protected function getKeysWithWhere($where)
             ->store('users')
             ->index($where['column'])->items();
 
+        if (in_array($where['column'], ['groups', 'roles'])) {
+            $items = $items->map(function ($item) {
+                return $item->map->handle()->values()->toArray();
+            });
+
+            if (is_array($where['values'])) {
+                $where['type'] = 'JsonContains';
+            }
+        }
+
```

Comment:
I might be missing the point of what this is doing, but I think rather than adding complexity to the query builder just yet, we should just adjust the query within the `users` tag here:

https://github.com/statamic/cms/blob/f6b573ee5046b55f75bb450c69fe2ec5529eca34/src/Tags/Users.php#L22-L28

Instead of querying `->where('group', $group)` (which is obviously wrong - it must be outdated code), we could instead use `->where('groups/groupname', true)`

Since you added `orWhere` support, we could loop over all the provided groups in the parameter.

```php
foreach ($this->params->explode('group', []) as $i => $group) {
    $method = $i === 0 ? 'where' : 'orWhere';
    $query->$method('groups/'.$group, true);
}
```

(That is untested - just wrote it on the fly in this comment)

Reply:
Ok I've updated the tag to work this way (we dont need to worry about the method approach as orWhere will not care).